cocos2d/CCScheduler.m

```diff
@@ -71,32 +76,190 @@ @interface CCTimer (Private)
 @end
 
 
-@interface CCScheduler (Private) <CCSchedulerTarget>
+@interface CCScheduler() <CCSchedulableTarget>
+
+@property(nonatomic, strong) CCTimer *timers;
+
 -(void)scheduleTimer:(CCTimer *)timer retain:(BOOL)retain;
+
 @end
 
 
-@implementation CCScheduledTarget {
-	__unsafe_unretained NSObject<CCSchedulerTarget> *_target;
-	CCTimer *_timers;
+//MARK: Copy on Write Arrays
+
+// This is an NSMutableArray-like class that implements copy-on-write to allow modifying an array while iterating it.
+// This is for performance reasons so that we don't need to copy large arrays every frame to iterate them safely.
+// Instead, the copy is only performed when the array is actually changed.
+
+@interface CopyOnWriteArray : NSObject<NSFastEnumeration> @end
```

Comment:
Hmm, maybe. It's a completely internal class though, only used within CCScheduler.m

Just as a matter of length of the file, CCTimer could be moved out. It is used externally. (and in user code)